presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -123,19 +129,35 @@ protected Node visitShowStats(ShowStats node, Void context)
             if (node.getRelation() instanceof TableSubquery) {
                 Query query = ((TableSubquery) node.getRelation()).getQuery();
                 QuerySpecification specification = (QuerySpecification) query.getQueryBody();
-                Plan plan = queryExplainer.get().getLogicalPlan(session, query(specification), parameters, warningCollector);
+                Plan plan = getPlanForSpecification(specification);
                 validateShowStatsSubquery(node, query, specification, plan);
                 Table table = (Table) specification.getFrom().get();
                 Constraint constraint = getConstraint(plan);
-                return rewriteShowStats(node, table, constraint);
+                return rewriteShowStats(node, table, specification.getSelect().getSelectItems(), constraint);
             }
             if (node.getRelation() instanceof Table) {
                 Table table = (Table) node.getRelation();
-                return rewriteShowStats(node, table, Constraint.alwaysTrue());
+                return rewriteShowStats(node, table, ImmutableList.of(new AllColumns()), Constraint.alwaysTrue());
             }
             throw new IllegalArgumentException("Expected either TableSubquery or Table as relation");
         }
 
+        private TableMetadata getTableMetadata(Node node, QualifiedName table)
+        {
+            return metadata.getTableMetadata(session, getTableHandle(node, table));
+        }
+
+        private Plan getPlanForSpecification(QuerySpecification specification)
+        {
+            try {
+                return queryExplainer.get().getLogicalPlan(session, query(specification), parameters, warningCollector);
+            }
+            catch (AccessDeniedException e) {
+                TableMetadata metadata = getTableMetadata(specification, ((Table) specification.getFrom().get()).getName());
```

Comment:
I am concerned about cast in catch clause, as this is an assumption.

On  the calling side, this assumption is already made in a form of a cast: `((Table) specification.getFrom().get())`.
So pass the `Table` object here.

Second, I think we can skip `getTableMetadata` here and use `Table#getName` for error reporting directly.

Reply:
Then we loose schema information (we will be only able to return table name). If it's ok - I will make it consistent

Reply:
> Then we loose schema information (we will be only able to return table name).

@kokosing @electrum opinions?